sparse/rest/handlers.go

```diff
@@ -103,6 +105,11 @@ func (server *SyncServer) doOpen(request *http.Request) error {
 	}
 	log.Infof("Receiving %v: size %v, directIO %v", server.filePath, interval.End, directIO)
 
+	if _, err := os.Stat(server.filePath); err != nil && errors.Is(err, os.ErrNotExist) {
+		log.Infof("file %v does not exist", server.filePath)
+		server.fileAlreadyExists = false
+	}
```

Comment:
Suggest moving this to the constructor method `Server` of SyncServer, so you don't need to calculate this here. For thread-safe purpose, even we know internally it will be only called once.

Reply:
Updated. Thank you.